### ISSUE TITLE
Linking nether portals 1.18

### ISSUE THREAD

**Issue by GroGaloraGran** [PaperSpigot/Paper#6995]
### Expected behavior

When entering a previously (in older versions) linked Nether portal, it should lead to the correct portal

### Observed/Actual behavior

Now the portal, which was previously Linked by coordinates, leads to the wrong portal, which is located nearby in Nether but at a different height

### Steps/models to reproduce

In our upper world, a gold farm was created on Nether portals, for which a "service" portal was left in Nether, which was Linked with the farm portals. Below in Nether and a little to the side there is a metro line with a portal through which you can get to the farm.
If we return from the upper world to Nether through the correct portal, we find ourselves in the service portal in Nether (this is dangerous due to the aggressive zombie pig)

### Plugin and Datapack List

ChestShop, 
ClearLag, 
Essentials, 
EssentialsChat, 
EssentialsProtect, 
EssentialsSpawn, 
LogBlock(do not work), 
LWC, 
MCLink*, 
Multiverse-Core, 
Multiverse-NetherPortals, 
PermissionsEx*, 
RandomTP,
Vault, 
WorldEdit, 
WorldEditSelectionVisualizer, 
WorldGuard

### Paper version

This server is running Paper version git-Paper-19 (MC: 1.18) (Implementing API version 1.18-R0.1-SNAPSHOT) (Git: 892c292)

### Other

If this is not an error, your hint is needed on how to correct the connection of portals

**Comment by Slitzo3** [comment]
I'm also experiencing this with Paper-22

**Comment by JamesRagnarok** [comment]
It might be a Multiverse issue. I was also experiencing this but got fixed after Multiverse removed.

**Comment by Machine-Maker** [comment]
Can someone with this issue, try loading up the world in vanilla and seeing if they have the same problem? Afaik, portals don’t actually “remember” what portal they were linked with, they just calculate where they were supposed to go and try to see if one exists nearby

**Comment by JamesRagnarok** [comment]
> Can someone with this issue, try loading up the world in vanilla and seeing if they have the same problem? Afaik, portals don’t actually “remember” what portal they were linked with, they just calculate where they were supposed to go and try to see if one exists nearby

Just tried vanilla and Spigot, not seeing this issue.

Edit: This issue still present with vanilla/Spigot + Muiltiverse, but fixed without Multiverse.

**Comment by e-im** [comment]
I haven't been able to reproduce this behaviour. Does anyone have a world where this does happen (currently on 1.17) and would be willing to upload it here? Is this issue reproducible without multiverse on paper?

**Comment by JamesRagnarok** [comment]
tl;dr: I felt like this is rather a Multiverse-NetherPortals issue.


Here's the map file: [portal_issue.zip](https://github.com/PaperMC/Paper/files/7644127/portal_issue.zip)

This was trimmed from my 1.17.1 map, which presents a [ilmango's shulker farm](https://youtu.be/RRSnya62snY). At the very bottom of both sides are nether portals function as a chunk loader, which should be connected (-231, 145, -780 at overworld, -29, 129, -97 at nether).

With Paper 1.17.1 build 391, it works pretty fine whether with or without Multiverse plugins. With Paper 1.18 build 22, after the maps upgraded, it still works without any plugins. But with Multiverse-Core **and Multiverse-NetherPortals**, going through the overworld side would end up at another upper portal in the nether side (-28, 184, -97).
 
Edit: I was using Multiverse-Core v4.3.1 and Multiverse-NetherPortals v4.2.1

**Comment by GroGaloraGran** [comment]
Yes, there is no problem with the disabled plugin Multiverse-NetherPortals v4.2.1, but we have an additional world with our own Nether world, and if we try to get from the additional world (let it be World2) and its world2_Nether with the plugin disabled, then for some reason we find ourselves in world_nether ((
Multiverse-Core v4.3.0 Multiverse-NetherPortals v4.2.1

**Comment by zdwolfe** [comment]
I can also report the same issue. Walking through a nether portal that was created in 1.17.1 from the Nether creates a new portal on the Overworld.

Plugins: SinglePlayerSleep, Paper. No other plugins/mods (others above reported Multiverse - I'm not using that).


```
[01:53:44 INFO]: This server is running Paper version git-Paper-29 (MC: 1.18) (Implementing API version 1.18-R0.1-SNAPSHOT) (Git: 675d1e3)
```

**Comment by electronicboy** [comment]
One thin with paper specifically, especially from older versions in which I don't recall when I fixed this, is the vanilla dimension scaling option in paper.yml, maybe worth checking that, but, portals have always been kinda weird as they don't actually link so there are cases in vanilla where this can occur

**Comment by JuanBadillo** [comment]
Can also report this same Issue. 
On a world generated on 1.17 walking through previously linked portals now directs to a new portal. Also worth nothing that the newly generated portals share coordinates on the Nether and Overworld, so a portal in X:20, Y:20 in Overworld would link to a portal in X:20 Y:20 in the nether, instead of scaling.

Tested with no active plugins, issue persists

**Comment by GroGaloraGran** [comment]
With the plugin disabled, teleportation in my case is fixed, although the first exit from Nether will mix the portal in the World to the side and spoil the buildings. Re-enabling the plugin returns the problem, but without it teleportation from world-2 to nethet-world-2 does not work and teleports to normal Nether is even worse. I wrote a parallel incident and in support of the plugin.
but as I see above they write that this does not fix the problem for everyone

**Comment by ptrblauw** [comment]
I fixed video in my ticket. https://github.com/PaperMC/Paper/issues/7051 ticket marked as duplicate of this ticket maybe it can help you to fix it.

**Comment by GroGaloraGran** [comment]
Hey
to confirm the anomaly, I made an additional video
Logs and videos using the Multiverse-NetherPortals plugin
https://youtu.be/OzTZgKUTZc4
https://gist.github.com/GroGaloraGran/75e7c5648cc5d3f7ef4bef29240b484c
Logs and videos with disabled plugin
https://youtu.be/DQRweOETx5s
https://gist.github.com/GroGaloraGran/78330ebcbe7425fe40277e0d9663551e
this is what is on our assembly at the moment, with the latest test core Paper 1.18.1 # 71

**Comment by thouu** [comment]
I'm also having the same issue on my server running Paper 76.

**Comment by thouu** [comment]
I'm also having this issue on Paper 77 for 1.18.1, there's screenshots of it [here](https://imgur.com/a/xhptEmr).
All these portals worked correctly before the 1.18.1 update, and now they make new portals instead of the ones that were used before.
My nether portal configuration is correct in `paper.yml`

When I tear down the newly-created portal, the old one starts working again and makes no new portals at all.

**Comment by zdwolfe** [comment]
I originally tried on 1.18 which had this issue on almost every portal so I scrapped the upgrade. I just recently tried 1.18.1 latest snap (Git: 5883773), and only one portal showed this behavior.

**Comment by Ghost-chu** [comment]
I can reproduce this issue without Multiverse-Core but PhantomWorlds.

**Comment by molor** [comment]
Can reproduce this in build 150 w/o any plugins that adds worlds or changes it. One of my players noticed that the portal from his home in overworld endlessly creates new ones in the nether in ~5-25 blocks nearby to previous, I checked and confirmed that it happens with me too.

**Comment by berlincount** [comment]
Seems to relate to Multiverse/Multiverse-NetherPortals#244

**Comment by Ghost-chu** [comment]
> Seems to relate to [Multiverse/Multiverse-NetherPortals#244](Multiverse/Multiverse-NetherPortals#244)

Multiverse/Multiverse-NetherPortals#244

**Comment by berlincount** [comment]
> > Seems to relate to [Multiverse/Multiverse-NetherPortals#244](Multiverse/Multiverse-NetherPortals#244)
> 
> [Multiverse/Multiverse-NetherPortals#244 (comment)](Multiverse/Multiverse-NetherPortals#244)

sure, but this still might mean that MultiVerse[-NetherPortals] and PhantomWorlds implement something (badly) the same way as well.

**Comment by Ghost-chu** [comment]
> > > Seems to relate to [Multiverse/Multiverse-NetherPortals#244](Multiverse/Multiverse-NetherPortals#244)
> > 
> > 
> > [Multiverse/Multiverse-NetherPortals#244 (comment)](Multiverse/Multiverse-NetherPortals#244)
> 
> sure, but this still might mean that MultiVerse[-NetherPortals] and PhantomWorlds implement something (badly) the same way as well.

how you explain
https://github.com/PaperMC/Paper/issues/6995#issuecomment-1013673759

MV main logic had no updates in past years.
If it can cause issue, that mean Bukkit API have bug and need to fix.

**Comment by berlincount** [comment]
> > > > Seems to relate to [Multiverse/Multiverse-NetherPortals#244](Multiverse/Multiverse-NetherPortals#244)
> > > [Multiverse/Multiverse-NetherPortals#244 (comment)](Multiverse/Multiverse-NetherPortals#244)
> > sure, but this still might mean that MultiVerse[-NetherPortals] and PhantomWorlds implement something (badly) the same way as well.
> 
> how you explain [#6995 (comment)](https://github.com/PaperMC/Paper/issues/6995#issuecomment-1013673759)
> 
> MV main logic had no updates in past years. If it can cause issue, that mean Bukkit API have bug and need to fix.

okay, fairynuff. maybe I'm low on coffee - I bow to your wisdom and invalidate my comments as meaningless!

**Comment by berlincount** [comment]
Is https://hub.spigotmc.org/jira/projects/SPIGOT/issues/ the right place where this should be reported?

I couldn't find anything there, i.e. I'm not sure this problem is known upstream ... is someone able to reproduce it with the upstream server version easily and could do a report, please?

**Comment by SmolFry** [comment]
Having this issue without any multiverse plugins. Portal in overworld creates one in nether, nether portal creates a different one in the overworld a few chunks away from the original.

Only plugins I have are Essentials, and Dynmap

**Comment by berlincount** [comment]
> Having this issue without any multiverse plugins. Portal in overworld creates one in nether, nether portal creates a different one in the overworld a few chunks away from the original.
> 
> Only plugins I have are Essentials, and Dynmap

hmmm, would you be able to test this with a Spigot server binary?

**Comment by berlincount** [comment]
Tried with Spigot per instructions from #7419, there everything is fine.
https://github.com/PaperMC/Paper/issues/7419#issuecomment-1029142258

**Comment by jpenilla** [comment]
should be fixed along with #7419 by #7438
